webapp/src/modules/item/sagas.ts

```diff
@@ -76,6 +89,26 @@ export function* itemSaga(getIdentity: () => AuthIdentity | undefined) {
   yield takeEvery(SET_PURCHASE, handleSetItemPurchaseWithCard)
   yield takeEvery(FETCH_ITEM_REQUEST, handleFetchItemRequest)
 
+  // to avoid race conditions, just one fetch items request is handled at once in the browse page
+  function* handleOneFetchItemsRequestAtOnce(): SagaIterator {
+    let task
+
+    while (true) {
+      const action: FetchItemsRequestAction = yield take(FETCH_ITEMS_REQUEST)
+      const {
+        pathname: currentPathname
+      }: ReturnType<typeof getLocation> = yield select(getLocation)
+
+      // if we have a task running in the browse path, we cancel the previous one
+      if (matchPath(currentPathname, { path: locations.browse() })) {
+        if (task && task.isRunning()) {
+          yield cancel(task)
+        }
+      }
```

Comment:
What do you think about merging the ifs here?
```suggestion
      if (matchPath(currentPathname, { path: locations.browse() }) && task && task.isRunning()) {
        if () {
          yield cancel(task)
        }
      }
```

Reply:
I have two questions about this: 
1. We do know what the current path is, but, how do we know if the last browse was performed from the `browse` path? I'm a bit in doubt if it matters if it was launched from another path.
2. The path when switching filters in the browse or collectibles page does not change (the query parameters do). Do we want to prevent race conditions when switching filters as well?